src/plugins/backend_odbc/backend_odbc_get.c

```diff
@@ -0,0 +1,842 @@
+ *
+ *
+ * This function is used to retrieve data that gets returned by the execution of a SELECT-statement and didn't fit into the
+ * fix-sized buffers. For such values, memory must be allocated dynamically.
+ *
+ * The function also re-allocates the buffer if it was still too small to get the whole returned data.
+ * This is especially useful for data sources that don't know the size of the returned data in advance.
+ *
+ * @pre The buffer @p targetValue must point to allocated memory of size @p bufferSize.
+ * Please consider that the buffer should be of type char*, like common strings, not of type char**, like string-arrays.
+ * The buffer is just passed as a pointer (char**) because the memory maybe gets reallocated and moved to a different memory address.
+ *
+ * @param sqlStmt The executed statement, for which SQLFetch() has been called before
+ * 	This handle gets freed if an error occurred, so don't dereference it if the function returned 'false'.
+ * @param colNumber The number of the column from which the data should be retrieved
+ * @param targetType The C data type that the fetched result should be converted to
+ * 	See https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/converting-data-from-c-to-sql-data-types for supported
+ * conversions.
+ * @param[in,out] targetValue The buffer where the data should be stored
+ * @param bufferSize The size of the buffer (as given to malloc())
+ * @param[out] errorKey Used to store errors and warnings
+ *
+ * @rerval 'true' if the data was fetched successfully
+ * @retval 'false' if an error occurred (see @p errorKey for details)
+ *
+ * @note The row number is not given as an argument, but related to the last call of SQLFetch()
+ *
+ * @see fetchResults() for processing the results of an executed SELECT statement
+ */
+static bool getLongData (SQLHSTMT sqlStmt, SQLUSMALLINT colNumber, SQLSMALLINT targetType, char ** targetValue, SQLLEN bufferSize,
+			 Key * errorKey)
+{
+	SQLLEN getDataLenOrInd;
+	SQLRETURN getDataRet;
+	unsigned int iteration = 0;
+
+	if (!sqlStmt)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "A NULL pointer was given as an argument for 'sqlStmt' to getLongData()");
+		return false;
+	}
+	else if (!targetValue)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey,
+					     "A NULL pointer was given as an argument for the buffer 'targetValue' to getLongData()");
+		return false;
+	}
+	else if (bufferSize <= 0)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "A buffer size <=0 was given as an argument to getLongData()");
+		return false;
+	}
+
+	do
+	{
+		if (iteration > 0)
+		{
+			if (elektraRealloc ((void **) targetValue, bufferSize * (iteration + 1)) == -1)
+			{
+				SQLFreeHandle (SQL_HANDLE_STMT, sqlStmt);
+				ELEKTRA_SET_OUT_OF_MEMORY_ERROR (errorKey);
+				return false;
+			}
+		}
+
+		getDataRet = SQLGetData (sqlStmt, colNumber, targetType, (*targetValue) + (iteration * (bufferSize - 1)), bufferSize,
+					 &getDataLenOrInd);
```

Comment:
Hm, the distinction is made via the return-value.
If `SQL success` is returned, the buffer was big enough.
The function is primary a helper for unknown value sizes, otherwise, just pass the known size.

The `nameLenOrInd' values inside the `struct columnData` are used in various places, so I used them for this case too to be consistent.

As the function is `static` and therefore only used internally, I think we can retain the current logic.